Civi/Financeextras/Hook/alterContent/PaymentStatus.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Civi\Financeextras\Hook\alterContent;
+
+class PaymentStatus {
+
+  private $callerPageTemplate;
+
+  private $callerPageContent;
+
+  /**
+   * The pages that are affected
+   * by this hook and their templates path.
+   */
+  const TARGET_PAGES_TEMPLATES = [
+    'contribution-form' => 'CRM/Contribute/Page/Tab.tpl',
+    'membership-form' => 'CRM/Member/Page/Tab.tpl',
+    'event-registration' => 'CRM/Event/Page/Tab.tpl',
+  ];
+
+  public function __construct($callerPageTemplate, &$callerPageContent) {
+    $this->callerPageTemplate = $callerPageTemplate;
+    $this->callerPageContent = &$callerPageContent;
+  }
+
+  public function run(): void {
+    if (!$this->shouldRun()) {
+      return;
+    }
+
+    $this->enforceAccountReceivablePaymentMethodOnPendingContribution();
+  }
+
+  private function shouldRun(): bool {
+    if (in_array($this->callerPageTemplate, self::TARGET_PAGES_TEMPLATES)) {
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Adds JS code that when Pending status for the contribution is selected,
+   * defaults the payment method to 'Account Receivable' enforce it
+   * at the user level by hiding the field.
+   *
+   * @return void
+   */
+  private function enforceAccountReceivablePaymentMethodOnPendingContribution(): void {
+    // JS variables are not available on contact page tabs when using something like this:
+    // https://docs.civicrm.org/dev/en/latest/framework/resources/#add-javascript-variables, so I had
+    // to assign them this way instead.
+    $helperVars = "<script>
+                     var pendingStatusId = {$this->getPendingStatusId()};
+                     var accountsReceivablePaymentMethodId = '{$this->getAccountsReceivablePaymentMethodId()}';
+                     var paymentDetailsSectionSelector = '{$this->getPaymentDetailsSectionSelector()}';
+                   </script>";
+    $this->callerPageContent .= $helperVars;
+
+    $url = \CRM_Core_Resources::singleton()->getUrl('io.compuco.financeextras', 'js/payment_status.js');
```

Comment:
Maybe better to use existing URL function  `E::url('js/payment_status.js')` in the `ExtensionUtil  `class? So we don't have to define extension key?